src/fd.cpp

```diff
@@ -111,12 +128,46 @@ static int wait_thread(void *arg) {
     }
   }
 
-  execLaterNative2(later_callback, static_cast<void *>(argsptr.release()), 0, args->loop);
+  callbackRegistryTable.scheduleCallback(later_callback, static_cast<void *>(argsptr.release()), 0, args->loop);
 
   return 0;
 
 }
 
+Rcpp::RObject execLater_fd_threaded(std::shared_ptr<ThreadArgs> args) {
+
+  std::unique_ptr<std::shared_ptr<ThreadArgs>> argsptr(new std::shared_ptr<ThreadArgs>(args));
+
+  tct_thrd_t thr;
+  if (tct_thrd_create(&thr, &wait_thread, static_cast<void *>(argsptr.release())) != tct_thrd_success)
+    Rcpp::stop("Thread creation failed");
+  tct_thrd_detach(thr);
+
+  Rcpp::XPtr<std::shared_ptr<std::atomic<bool>>> xptr(new std::shared_ptr<std::atomic<bool>>(args->flag), true);
+
+  return Rcpp::RObject(xptr);
```

Comment:
Is this necessary or can you just return xptr?

Reply:
The best I managed is this implicit SEXP conversion in [95adc95](https://github.com/shikokuchuo/later/pull/1/commits/95adc95f8ed49b8a928b9aad86c7a766558a1bc4), which I think is practically the same.

It works fine to return the `Rcpp::XPtr<std::shared_ptr<std::atomic<bool>>>` keeping that type all the way to the exported function, but then the automated Rcpp generator fails.

Reply:
Slightly cleaner way in [be661aa](https://github.com/shikokuchuo/later/pull/1/commits/be661aaa1541a99837ff7448812c2af19166f3ec).